bin/runtime/src/chain_extension/baby-liminal-extension/lib.rs

```diff
@@ -87,6 +87,15 @@ pub trait BabyLiminalExtension {
         input: Vec<u8>,
         system: ProvingSystem,
     );
+
+    #[ink(extension = 43, returns_result = false, handle_status = false)]
+    fn poseidon_one_to_one(input: [[u64; 4]; 1]) -> [u64; 4];
+
+    #[ink(extension = 44, returns_result = false, handle_status = false)]
+    fn poseidon_two_to_one(input: [[u64; 4]; 2]) -> [u64; 4];
```

Comment:
why like this, why not one, two and four arguments respectively?

Reply:
so that decoding on the chain side is easier :P

unfortunately, when you implement `ChainExtension` and have `env` object, you cannot read arguments sequentially (the pointer is not moved); that sucks ofc